git-protocol/src/fetch/response/mod.rs

```diff
@@ -108,10 +119,25 @@ impl Acknowledgement {
     }
 }
 
+impl WantedRef {
+    /// Parse a `WantedRef` from a `line` as received from the server.
+    pub fn from_line(line: &str) -> Result<WantedRef, Error> {
+        match line.trim_end().split_once(' ') {
+            Some((id, path)) => {
+                let id =
+                    git_hash::ObjectId::from_hex(id.as_bytes()).map_err(|_| Error::UnknownLineType(line.to_owned()))?;
+                Ok(WantedRef { id, path: path.into() })
+            }
+            _ => unreachable!("cannot have an entirely empty line"),
```

Comment:
Ehm yes. I'm not sure about those `unreachable!`s here -- should they not better be `Err` results, just in case you're talking to something which is not actually a git server?

Reply:
Agreed, since it's untrusted input it's probably a better idea to not crash the client process because somebody isn't conforming to the protocol.